packages/engine/src/vaults/impl/sol/Vault.ts

```diff
@@ -361,9 +361,9 @@ export default class Vault extends VaultBase {
 
   override validateAddress(address: string): Promise<string> {
     try {
-      if (PublicKey.isOnCurve(address)) {
-        return Promise.resolve(address);
-      }
+      // eslint-disable-next-line no-new
+      new PublicKey(address);
+      return Promise.resolve(address);
```

Comment:
solana 的账户地址是否合法应该由 PublicKey.isOnCurve 函数来校验，因为 Keypaid 生成的地址总是会位于 ed25519 曲线上。

这个 issue 要解决的是 Token 地址不合法的问题，有单独的 vault method 来处理，不用改动账户地址校验的方法


```suggestion
  override validateAddress(address: string): Promise<string> {
    try {
      if (PublicKey.isOnCurve(address)) {
        return Promise.resolve(address);
      }
    } catch {
      // pass
    }
    return Promise.reject(new InvalidAddress());
  }

  override async validateTokenAddress(address: string): Promise<string> {
    try {
      return await this.validateAddress(address);
    } catch {
      const tokenAddress = new PublicKey(address);
      if (tokenAddress) {
        return address;
      }
      throw new InvalidAddress();
    }
  }
```

Reply:
https://stackoverflow.com/questions/71200948/how-can-i-validate-a-solana-wallet-address-with-web3js

Reply:
了解了

Reply:
或者简单点 validateTokenAddress 直接 new PublicKey 来判断即可

Reply:
okay 改好啦 😆

Reply:
这个地方这样做还是不太稳妥，现在的做法校验ata可能还行，但是mint address还是用api更稳妥 `getAccountInfo`, 看下响应中的owner字段是不是 `TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA`

Reply:
跟 @somebodyLi  讨论了下，这块维持 new PublicKey 的判断就可以了